h2o-core/src/main/java/water/server/ExtensionHandler.java

```diff
@@ -0,0 +1,44 @@
+package water.server;
+
+import water.AbstractH2OExtension;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Extension point for HTTP request handling.
+ */
+public interface ExtensionHandler {
+  /**
+   * Extended request handler
+   * @param target -
+   * @param request -
+   * @param response -
+   * @return true if the request should be processed handled, false otherwise
+   * @throws IOException -
+   * @throws ServletException -
+   */
+  boolean handle(String target, HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException;
+
+  /**
+   * Static registry of handler extensions.
+   * Customizations can use this to inject their handlers here during {@link AbstractH2OExtension#init()}.
+   */
+  Registry REGISTRY = new Registry();
```

Comment:
I think it would be better to use SPI to register extensions to be consistent.

Reply:
which SPI? ServiceLoader? it is already used to load extensions, from which this is called

Reply:
ok done